src/components/form-editor/FormEditor.tsx

```diff
@@ -39,19 +57,35 @@ const FormEditor = ({
         Basic information
       </Typography>
       {DEFAULT_STUDENT_FIELDS.concat(DEFAULT_FAMILY_FIELDS).map((field) => (
-        <FieldEditor key={field.id} field={field} isReadOnly={isReadOnly} />
+        <FieldEditor
+          key={field.id}
```

Comment:
why don't we add `existingId` as a prop here? There's some logic in the `FieldEditor` that checks if a field is Default or has an existing id, but it seems like it's just returning the id of the Field in both cases. Do you mind  briefly explaining why?

Reply:
the value of `existingId` determines whether we're going to create or edit a field when users click submit! if it's null, then we make a post request, and otherwise we make a put.

I'll rename this to `existingFieldId` which can hopefully be a little clearer?

Reply:
that makes sense, I'll leave the naming up to you. I'm fine either way